src/MICore/Debugger.cs

```diff
@@ -604,14 +607,49 @@ protected bool IsCoreDump
             }
         }
 
+        [DllImport("kernel32.dll", SetLastError = true)]
+        public static extern IntPtr OpenProcess(int dwDesiredAccess, [MarshalAs(UnmanagedType.Bool)] bool bInheritHandle, int dwProcessId);
+        [DllImport("kernel32.dll", SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        static extern bool TerminateProcess(IntPtr hProcess, uint uExitCode);
+
         public async Task<Results> CmdTerminate()
         {
             if (!_terminating)
             {
                 _terminating = true;
-                if (ProcessState == ProcessState.Running && this.MICommandFactory.Mode != MIMode.Clrdbg)
-                {
-                    await AddInternalBreakAction(() => MICommandFactory.Terminate());
+                if (ProcessState == ProcessState.Running &&
+                    this.MICommandFactory.Mode != MIMode.Clrdbg)
+                {
+                    // MinGW and Cygwin on Windows don't support async break. Because of this,
+                    // the normal path of sending an internal async break so we can exit doesn't work.
+                    // Therefore, we will call TerminateProcess on the debuggee with the exit code of 0
+                    // to terminate debugging. 
+                    if ((this.IsCygwin || this.IsMinGW) && _debuggeePids.Count > 0)
+                    {
+                        int debuggeePid = _debuggeePids.First().Value;
```

Comment:
Doesn't hurt to terminate all pids. Normally having more than one is a transient state. However there is a non-trivial amount of work done by gdb in between reporting that a new process is available and detaching so that vs can reattach separate miengine instance. If something goes wrong/hangs then killing all children is most likely the right way to recover.